exp/ipam/internal/webhooks/ipaddress.go

```diff
@@ -122,17 +122,35 @@ func (webhook *IPAddress) validate(ctx context.Context, ip *ipamv1.IPAddress) er
 				"prefix is too large for an IPv6 address",
 			))
 	}
-
-	_, err = netip.ParseAddr(ip.Spec.Gateway)
-	if err != nil {
+	if addr.Is4() && ip.Spec.Gateway == "" {
```

Comment:
Just reading through this, are we not specifying a mode of operation? Wouldn't it be possible in this way to mix ipv4 and ipv6 settings under the same specs?

Reply:
There isn't anything explicit that specifies the mode of operation on the IPAddress, but as part of this PR we did add validation to ensure address and gateway fields are not [mixed](https://github.com/kubernetes-sigs/cluster-api/pull/8525/files/792e5667d4b6777bb1052924d364a51ff200ccff#diff-55959ca63fd3777ed769c3ac33612c5f95d60a762a748300f1f0f1a468e865f2R144), which I think would be sufficient to prevent mixing ipv4 and ipv6 on the same IPAddress.